src/Microsoft.OData.Core/ODataLibraryCompatibility.cs

```diff
@@ -4,29 +4,49 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using System.Diagnostics.CodeAnalysis;
+using System;
 
 namespace Microsoft.OData
 {
     /// <summary>
     /// Library compatibility levels.
     /// </summary>
-    [SuppressMessage("Microsoft.Design", "CA1008:EnumsShouldHaveZeroValue")]
+    [Flags]
     public enum ODataLibraryCompatibility
     {
         /// <summary>
-        /// Version 6.x
+        /// No backward compatibility feature flag set.
         /// </summary>
-        Version6 = 060000,
+        None = 0,
 
         /// <summary>
-        /// Version 7.x
+        /// When enabled, the CSDL Scale and SRID "variable" value will be written "Variable" instead of "variable" for compatibility with older versions of the library.
+        /// </summary>
+        UseLegacyVariableCasing = 1 << 0,
+
+        /// <summary>
+        /// When enabled, writes a null odata annotation.@odata.null=true

```

Comment:
```suggestion
        /// When enabled, writes '@odata.null=true' if a top-level property is single-valued and has null value.
```
That is the behaviour we're trying to vary if this [OData V3 compatibility](https://learn.microsoft.com/en-us/openspecs/odata_standards/ms-odatajson/47e36f56-b0a6-4305-befa-3a797bd3f5a1) flag is enabled, otherwise a `204 No Content` is what should be returned -  [11.2.3](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/redlined/part1-protocol/odata-v4.0-errata03-os-part1-protocol-redlined.html#_Toc447291031)